contracts/core/Allo.sol

```diff
@@ -513,24 +427,72 @@ contract Allo is Transfer, Initializable, Ownable, AccessControl {
     }
 
     /// ====================================
-    /// ======= Internal Functions =========
+    /// ======= Strategy Functions =========
     /// ====================================
 
+    /// @notice passes _data through to the allocation strategy for that pool
+    /// @notice returns the recipientId from the allocation strategy
+    /// @param _poolId id of the pool
+    /// @param _data encoded data unique to the allocation strategy for that pool
+    function registerRecipients(uint256 _poolId, bytes memory _data) external payable returns (address) {
+        return pools[_poolId].strategy.registerRecipients(_data, msg.sender);
+    }
+
+    /// @notice Fund a pool
+    /// @param _poolId id of the pool
+    /// @param _amount extra amount of the token to be deposited into the pool
+    /// @param _token The address of the token that the pool is denominated in
+    /// @dev Anyone can fund a pool
+    function fundPool(uint256 _poolId, uint256 _amount, address _token) external payable {
+        if (_amount == 0) {
+            revert NOT_ENOUGH_FUNDS();
+        }
+        _fundPool(_token, _amount, _poolId, pools[_poolId].strategy);
+    }
+
     /// @notice passes _data & msg.sender through to the allocation strategy for that pool
     /// @param _poolId id of the pool
     /// @param _data encoded data unique to the allocation strategy for that pool
+    function allocate(uint256 _poolId, bytes memory _data) external payable {
+        _allocate(_poolId, _data);
+    }
+
+    /// @notice vote to multiple pools
+    /// @param _poolIds ids of the pools
+    /// @param _datas encoded data unique to the allocation strategy for that pool
+    function batchAllocate(uint256[] calldata _poolIds, bytes[] memory _datas) external {
+        for (uint256 i = 0; i < _poolIds.length;) {
```

Comment:
we should cache the array length to save some gas
```
uint526 length = _poolIds.length;
for (uint256 i = 0; i < length;) { ... }
```

and we need to add an input validation (`_poolId.length == _data.length`)